client-apis/microprofile-restclient/src/main/java/org/wildfly/swarm/microprofile/restclient/BuilderImpl.java

```diff
@@ -102,13 +104,33 @@ public RestClientBuilder baseUrl(URL url) {
 
         ClassLoader classLoader = aClass.getClassLoader();
 
-        final T actualClient = this.builderDelegate.build()
-                .target(this.baseURI)
-                .proxyBuilder(aClass)
-                .classloader(classLoader)
-                .defaultConsumes(MediaType.TEXT_PLAIN)
-                .defaultProduces(MediaType.TEXT_PLAIN)
-                .build();
+        List<String> noProxyHosts = Arrays.asList(
+                System.getProperty("http.nonProxyHosts", "localhost|127.*|[::1]").split("|"));
+
+        final T actualClient;
+
+        if (System.getProperty("http.proxyHost") != null
+                && !noProxyHosts.contains(this.baseURI.getHost())) {
+            // Use proxy, if defined
+            actualClient = this.builderDelegate.defaultProxy(
```

Comment:
My only comment would be whether the System.getProperty() call result should be stored as a local variable as opposed to making two separate calls?

Don't recall if there's much overhead to System.getProperty() calls, just a thought

Reply:
Don't know either but changed the code accordingly. Just pushed it to the pull request.